### PR TITLE
fix(terraform): add aws_transfer_server to CKV2_AWS_5 check

### PR DESCRIPTION
**Pull request by Teko012**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

It adds `aws_transfer_server` to `CKV2_AWS_5` check in terraform, as it can use security groups. It also fixes some formatting issues in the same file, so the formatter can pass. 

Fixes https://github.com/bridgecrewio/checkov/issues/4136

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/graph_checks/aws/SGAttachedToResource.yaml [hunk 1]
```diff
@@ -37,18 +37,19 @@ definition:
         - aws_launch_configuration
         - aws_launch_template
         - aws_lb
+        - aws_memorydb_cluster
+        - aws_mq_broker
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/aws/SGAttachedToResource.yaml:41]: it looks like `aws_mq_broker` is twice there. thanks for reording, didn't see that before 😅

Comment (Teko012) [reply checkov/terraform/checks/graph_checks/aws/SGAttachedToResource.yaml:41]: Oh, indeed, let me remove it.

**Review by gruebel — APPROVED** [review]
great work and thanks for your first contribution 🎉

**Review by marynaKK — APPROVED** [review]
🎄 🍒

**Pull request merged**